module/zfs/dmu_recv.c

```diff
@@ -2881,7 +3191,7 @@ dmu_recv_stream(dmu_recv_cookie_t *drc, offset_t *voffp)
 	struct receive_writer_arg *rwa = kmem_zalloc(sizeof (*rwa), KM_SLEEP);
 
 	if (dsl_dataset_is_zapified(drc->drc_ds)) {
-		uint64_t bytes;
+		uint64_t bytes = 0;
```

Comment:
this was the fix for the FreeBSD `redacted_panic.ksh` test panic. It turns out the panic wasn't related to my patch, but the patch did happen to trigger the issue more reliably.

Big thanks to @PaulZ-98 for tracking down the cause of the panic. All tests are now passing.